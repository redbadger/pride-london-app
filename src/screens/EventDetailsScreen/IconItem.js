// @flow
import React from "react";
import { Image, View, StyleSheet } from "react-native";
import type { Node } from "react";
import Touchable from "../../components/Touchable";

type IconItemProps = {|
  children: Node,
  onPress?: Function,
  source: Image.propTypes.source
|};

const IconItem = ({ children, onPress, source }: IconItemProps) => {
  if (onPress) {
    return (
      <View style={styles.container}>
        <Image source={source} style={styles.icon} />
        <Touchable onPress={onPress} style={styles.item}>
          {children}
        </Touchable>
      </View>
    );
  }
  return (
    <View style={styles.container}>
      <Image source={source} style={styles.icon} />
      <View style={styles.item}>{children}</View>
    </View>
  );
};

IconItem.defaultProps = {
  onPress: undefined
};

// Note: minHeight, margin and paddings are to accomodate touchables. This
// component is designed to work as children of IconList which removes
// the outside margins.
const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "row",
    minHeight: 44
  },
  icon: {
    width: 32,
    height: 32,
    marginRight: 20,
    marginVertical: 6
  },
  item: {
    flexShrink: 1,
    justifyContent: "center",
    paddingVertical: 6
  }
});

export default IconItem;
