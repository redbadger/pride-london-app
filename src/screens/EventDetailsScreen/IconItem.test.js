// @flow
import React from "react";
import { Text } from "react-native";
import { shallow } from "enzyme";
import IconItem from "./IconItem";

it("renders correctly", () => {
  const output = shallow(
    <IconItem source={{ uri: "/super-awesome-image.jpg" }}>
      <Text type="h4">some thing</Text>
    </IconItem>
  );
  expect(output).toMatchSnapshot();
});

it("renders correctly when passed an onPress handler", () => {
  const output = shallow(
    <IconItem onPress={() => {}} source={{ uri: "/super-awesome-image.jpg" }}>
      <Text type="h4">some thing</Text>
    </IconItem>
  );
  expect(output).toMatchSnapshot();
});
