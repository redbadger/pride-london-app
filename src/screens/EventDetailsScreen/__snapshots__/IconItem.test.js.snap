// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Component
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "minHeight": 44,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "/super-awesome-image.jpg",
      }
    }
    style={
      Object {
        "height": 32,
        "marginRight": 20,
        "marginVertical": 6,
        "width": 32,
      }
    }
  />
  <Component
    style={
      Object {
        "flexShrink": 1,
        "justifyContent": "center",
        "paddingVertical": 6,
      }
    }
  >
    <Component
      type="h4"
    >
      some thing
    </Component>
  </Component>
</Component>
`;

exports[`renders correctly when passed an onPress handler 1`] = `
<Component
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "minHeight": 44,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "/super-awesome-image.jpg",
      }
    }
    style={
      Object {
        "height": 32,
        "marginRight": 20,
        "marginVertical": 6,
        "width": 32,
      }
    }
  />
  <Touchable
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    delayPressIn={50}
    onPress={[Function]}
    style={
      Object {
        "flexShrink": 1,
        "justifyContent": "center",
        "paddingVertical": 6,
      }
    }
  >
    <Component
      type="h4"
    >
      some thing
    </Component>
  </Touchable>
</Component>
`;
